runtime/lua/vim/lsp/util.lua

```diff
@@ -759,9 +759,11 @@ function M.rename(old_fname, new_fname, opts)
   vim.fn.bufload(oldbuf)
 
   -- The there may be pending changes in the buffer
-  api.nvim_buf_call(oldbuf, function()
-    vim.cmd('w!')
-  end)
+  if vim.fn.isdirectory(old_fname) == 0 then
```

Comment:
Should this check whether 'buftype' is set on the old buffer? E.g. could the old buffer be a :terminal or other non-file?

Reply:
So far as I know, LSP can rename only regular files and directories.